pages/portal/journal/review.tsx

```diff
@@ -2,49 +2,207 @@ import { NextPage,  NextPageContext } from "next";
+    const [responseStatus, setResponseStatus] = useState(0); //Used to display a success/error message.
+    
+
+
+    const handleEntryApproval = async (e:React.SyntheticEvent) => {
+        e.preventDefault();
+        let response;
+        let submitButton = e.target as HTMLButtonElement;
+        if(submitButton.name === "approve"){
+            //Approve the entry...
+            response = await fetch(`/api/journal/review?approved=true&id=${submitButton.value}`, {
+                method: "PUT",
+            });
+            setResponseStatus(response.status);
+
+        }
+
+        //Button by the approve button
+        if(submitButton.name === "rejecting"){
+            //If the user has viewed the warning modal and selects the option to not view it again, then the next time they click the reject button, the entry will be rejected.
+            if(!isRejecting && warningDismissed){
+                response = await fetch(`/api/journal/review?approved=false&id=${submitButton.value}`, {
+                    method: "PUT",
+                });
+                setResponseStatus(response.status);
+
+            } else {
+                setIsRejecting(true);
+                setRejectingID(submitButton.value);
+            }
+        }
+
+        //This button is within the reject modal. If a user chooses to not dismiss the warning, then they can click the reject button within it to reject the entry.
+        if(submitButton.name === "reject"){
+            //Reject here...
+            setIsRejecting(false);
+            response = await fetch(`/api/journal/review?approved=false&id=${currentRejectingID}`, {
+                method: "PUT",
+            });
+            setResponseStatus(response.status);
+        } 
+    };
+
+    const toggleWarningDismissed = (e:React.SyntheticEvent) => {
+        setWarningDismissed(true);
+    }
+
+
+
     return(
         <main className="wrapper">
             <Head>
                 <title>Review Journal Entries | Mindversity Website</title>
             </Head>
+            {isRejecting && (
+                <div className="rejectModal">
+                    <div className="modalBody">
+                        <h1>Are you sure?</h1>
+                        <p>Continuing with this action will delete the entry permanently.</p>
+                        <input type='checkbox' onClick={toggleWarningDismissed}/>
+                        <span>Do not show this message again.</span>
+                        <div className="actionButtonContainer">
+                            <button type="submit" name="reject" className="actionButton actionButtonPrimary" onClick={handleEntryApproval}>Reject</button>
+                            <button className="actionButton actionButtonSecondary" onClick={() => {setIsRejecting(false); setRejectingID("")}}>Close</button>
+                        </div>
+                    </div>
```

Comment:
I like the idea of the confirmation modal to make sure people don't accidentally reject the wrong one